test/fetch-test-deps.sh

```diff
@@ -1,10 +1,50 @@
 #!/usr/bin/env bash
-
 set -e
 
 cd "$(dirname "$0")"
 
-case "$1" in
+# Parse arguments and provide usage message #########################
+
+# Provide a usage message
+usage() {
+	echo "Downloads source code of Game Boy programs used as RGBDS test cases."
+	echo "Options:"
+	echo "    -h, --help      show this help message"
+	echo "    --only-free     download only freely licensed codebases"
+	echo "    --get-hash      print programs' commit hashes instead of downloading them"
+	echo "    --get-paths     print programs' GitHub paths instead of downloading them"
+}
+
+# Parse options in pure Bash because macOS `getopt` is stuck
+# in what util-linux `getopt` calls `GETOPT_COMPATIBLE` mode
+with_nonfree=true
+actionname=
+while [[ $# -gt 0 ]]; do
+	case "$1" in
+		-h|--help)
+			usage
+			exit 0
+			;;
+		--only-free)
+			with_nonfree=false
+			;;
+		--get-hash|--get-paths)
+			actionname="$1"
+			;;
+		--)
+			break
+			;;
+		*)
+			echo "$(basename $0): unknown option "$1""
+			exit 1
+			;;
+	esac
+	shift
+done
+
+# Act on arguments that have been accepted ##########################
+
```

Comment:
```suggestion
```

Honestly, if I didn't understand the code below, this comment wouldn't clarify anything. "Acting on arguments" is what essentially every shell script does.